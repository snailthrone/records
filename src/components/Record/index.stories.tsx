import React from 'react';
import {Â Meta, Story } from '@storybook/react/types-6-0';

import Record, { Props } from '.';

export default { title: 'Components/Record', component: Record } as Meta;

const Template: Story<Props> = args => <Record {...args} />

export const Default = Template.bind({});
Default.args = {
  title: 'Heavy as a Really Heavy Thing',
  performer: 'Strapping Young Lad',
  releaseDate: 1995,
  catalogId: '77092-2'
}
