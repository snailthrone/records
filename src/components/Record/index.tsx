import React, { FC } from 'react';

export type Props = {
  title: string;
  performer: string;
  releaseDate: number;
  catalogId?: string;
};

const Record: FC<Props> = props => {
  const { title, performer, releaseDate, catalogId } = props;
  return (
    <div>
      <h3>{title}</h3>
      <h4>{performer}</h4>
      <div>
        <span>{releaseDate}</span> { catalogId && <span>{catalogId}</span> }
      </div>
    </div>
  )
}

export default Record;
