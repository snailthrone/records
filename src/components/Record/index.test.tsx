import React from 'react';
import renderer from 'react-test-renderer';

import Record from './index'

const mockRecord = {
  title: 'Heavy as a Really Heavy Thing',
  performer: 'Strapping Young Lad',
  releaseDate: 1995,
  catalogId: '77092-2'
}

describe('Record', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Record {...mockRecord} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
