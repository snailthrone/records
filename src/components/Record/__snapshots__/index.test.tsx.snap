// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Record renders correctly 1`] = `
<div>
  <h3>
    Heavy as a Really Heavy Thing
  </h3>
  <h4>
    Strapping Young Lad
  </h4>
  <div>
    <span>
      1995
    </span>
     
    <span>
      77092-2
    </span>
  </div>
</div>
`;
